web-jehlomat/src/screens/EditaceUzivatele/EditaceUzivatele.tsx

```diff
@@ -0,0 +1,187 @@
+    const history = useHistory();
+    const isDesktop = useMediaQuery(media.gt('mobile'));
+    const { userId } = useParams<IParams>();
+    const loggedUser = useRecoilValue(userState);
+    const [user, setUser] = useState<IUser>();
+    const [teams, setTeams] = useState<ITeam[]>([]);
+    const [organization, setOrganization] = useState<IOrganizace>();
+
+    useEffect(() => {
+        if (token && userId) {
+            const loadData = async (token: string) => {
+                try {
+                    const userResponse: AxiosResponse<IUser> = await API.get(apiURL.getUser(userId));
+                    setUser(userResponse.data);
+                    if (userResponse.data.organizationId) {
+                        const teamsInOrganization: AxiosResponse<ITeam[]> = await API.get(apiURL.getTeamsInOrganization(userResponse.data.organizationId));
+                        setTeams(teamsInOrganization.data);
+                        const organizationInfo: AxiosResponse<IOrganizace> = await API.get(apiURL.getOrganization(userResponse.data.organizationId));
+                        setOrganization(organizationInfo.data);
+                    }
+                } catch (err) {
+                    history.push(LINKS.ERROR);
+                }
+                const response: AxiosResponse<IUser> = await API.get(apiURL.getUser(userId));
+                return response.data;
+            };
+            loadData(token);
+        }
+    }, [history, token, userId]);
+
+    if (!loggedUser?.isAdmin) {
+        return <AccessDenied />;
+    }
+
+    return (
+        <>
+            <Header mobileTitle="Editace uživatele" />
+            <Container
+                maxWidth="lg"
+                sx={{
+                    flexGrow: 1,
+                    display: 'flex',
+                    flexDirection: 'row',
+                    flexWrap: 'wrap',
+                }}
+            >
+                <Grid xs={isDesktop ? 7 : 12} item alignItems="start" container direction="column">
+                    {isDesktop && (
+                        <PageHeading align="left" variant="h1" color={primaryDark} sx={{ mt: '80px', mb: '86px', ml: '97px' }}>
+                            Editace uživatele
+                        </PageHeading>
+                    )}
+                    {user && (
+                        <FormWrapper className={isDesktop ? '' : 'mobile'}>
+                            <Formik
+                                initialValues={{
+                                    email: user.email,
+                                    username: user.username,
+                                    organizationId: user.organizationId,
+                                    teamId: user.teamId,
+                                }}
+                                validationSchema={validationSchema}
+                                onSubmit={async (values: IValues, { setErrors }: FormikHelpers<IValues>) => {
```

Comment:
Personally I like to avoid having complex code in the template section of component - better discoverability.